docs/dev/raft-topology.md

```diff
@@ -0,0 +1,115 @@
+Topology change algorithm over raft with failure recovery
```

Comment:
If you're writing python-like pseudocode, and using "```python ... ```" blocks for syntax coloring, then please use the right comment markers (I see sometimes there's #, sometime's there's //, let's use # consistently), and replace `version++` with `version += 1`